File: src/main/java/com/learnspark/submission/entity/Submission.java
package com.learnspark.submission.entity;

import com.learnspark.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Map;
import java.util.UUID;

/**
 * 任务提交记录实体，对应 submissions 表。
 *
 * <p>记录用户对某次任务的提交内容以及 AI 审核结果。
 * 一次提交对应一条记录；未通过可再次提交（生成新记录）。
 */
@Entity
@Table(name = "submissions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Submission extends BaseEntity {

    @Id
    @Column(name = "id", updatable = false, columnDefinition = "VARCHAR(36)")
    private String id;

    @Column(name = "task_id", nullable = false, columnDefinition = "VARCHAR(36)")
    private String taskId;

    @Column(name = "user_id", nullable = false, columnDefinition = "VARCHAR(36)")
    private String userId;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    /** 附件 URL 列表，存为 JSON 数组 */
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "attachment_urls", columnDefinition = "JSON")
    private List<String> attachmentUrls;

    @Column(name = "ai_feedback", columnDefinition = "TEXT")
    private String aiFeedback;

    @Column(name = "ai_score", columnDefinition = "SMALLINT")
    private Integer aiScore;

    @Column(name = "passed", columnDefinition = "TINYINT(1)")
    private Boolean passed;

    @Column(name = "ai_model", length = 100)
    private String aiModel;

    /** AI 原始响应，存为 JSON 对象 */
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "ai_raw_response", columnDefinition = "JSON")
    private Map<String, Object> aiRawResponse;

    @Column(name = "submitted_at")
    private LocalDateTime submittedAt;

    @Column(name = "reviewed_at")
    private LocalDateTime reviewedAt;

    @PrePersist
    void fillDefaults() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
        if (this.submittedAt == null) {
            this.submittedAt = LocalDateTime.now();
        }
    }
}

File: learnspark-kmp/server/src/main/kotlin/com/learnspark/server/domain/entity/Submission.kt
package com.learnspark.server.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

/**
 * 任务提交实体（按文档 §11.2.3：submissions 表）。
 *
 * 字段映射：旧版 content_md → 新版 content（统一字段命名）。
 */
@Entity
@Table(name = "submissions")
class Submission(
    @Id
    var id: String = "",

    @Column(name = "task_id", nullable = false)
    var taskId: String = "",

    @Column(name = "user_id", nullable = false)
    var userId: String = "",

    @Column(nullable = false, columnDefinition = "MEDIUMTEXT")
    var content: String = "",

    @Column(name = "ai_score")
    var aiScore: Int? = null,

    @Column(name = "ai_feedback", columnDefinition = "TEXT")
    var aiFeedback: String? = null,

    @Column(name = "ai_highlights", columnDefinition = "TEXT")
    var aiHighlights: String? = null,

    @Column(name = "reviewed_at")
    var reviewedAt: Instant? = null,

    @Column(nullable = false)
    var version: Long = 0,

    @Column(name = "created_at", insertable = false, updatable = false)
    var createdAt: Instant? = null,

    @Column(name = "updated_at", insertable = false, updatable = false)
    var updatedAt: Instant? = null,
)
